core/src/avm2/globals/flash/display/display_object.rs

```diff
@@ -541,14 +540,21 @@ pub fn set_rotation<'gc>(
 
 /// Implements `name`'s getter.
 pub fn get_name<'gc>(
-    activation: &mut Activation<'_, 'gc>,
+    _activation: &mut Activation<'_, 'gc>,
     this: Value<'gc>,
     _args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
     let this = this.as_object().unwrap();
 
     if let Some(dobj) = this.as_display_object() {
-        return Ok(dobj.name().unwrap_or_else(|| istr!("")).into());
+        if let Some(name) = dobj.name() {
+            return Ok(name.into());
+        } else {
+            // The `Stage` is the only DisplayObject that can be accessed in
+            // AVM2 that has a `None` name
+            assert!(matches!(dobj, DisplayObject::Stage(_)));
```

Comment:
Can we change it to a debug assert at least? Or add a path for stage? When it happens in production we shouldn't crash Ruffle